### DISCUSSION TITLE
Ignore/exclude an entrypoint or configuration from WDS HMR handling?

### DISCUSSION THREAD

**Discussion by kamranayub** [webpack/webpack#17816] [Q&A]
We are building up our new docs site with Docusaurus for our TypeScript game engine library. The library itself is packaged up with Webpack as both a UMD module and native ES modules (two separate builds).

Since we are using Docusaurus, that also uses Webpack and serves the local docs under webpack-dev-server.

The use case I'm trying to accomplish is to ensure that any live playgrounds (StackBlitz) are running against the _local changes_ to our library to smooth out the developer experience. That way, if you are making a change locally, and you adjust an API or add a new feature, you can write the docs and see your local changes reflected in all the playground examples. The alternative is that the playgrounds have to reference stable or tagged npm packages which won't reflect any local changes, causing a two-phase kind of workflow.

So what I'm trying to do with the Docusaurus webpack config is build a _local_ production copy (or dev, if I can make it work) of the library while running the docs so that we can pass the local build `.js` file to a sandbox playground (StackBlitz).

The way I've tried to accomplish this is through a separate entrypoint as well as a MultiCompiler config. Both approaches can work during a _build_ but not under WDS. It's very similar to [this issue](https://github.com/webpack/webpack-dev-server/issues/4469).

![image](https://github.com/webpack/webpack/assets/563819/d2c63658-ad75-4a74-b30c-7bdf7b24f0b2)

The issue is that during development, the library `.js` file includes all the WDS/HMR runtime stuff. ~If I understand correctly, that doesn't allow the `.js` file to be a standalone bundle I can just include on a page or reference in an import (outside of Webpack, in Stackblitz). It's instead meant to be used/loaded by the runtime, which isn't what I want when I send the contents of the file over to Stackblitz.~

**Update 11/21:** I actually discovered that the WDS build _does_ work standalone in Stackblitz, but [only if you fork the project](https://github.com/stackblitz/core/issues/2850) in the embedded version. This leads me to believe it's actually a Stackblitz compiler issue, not WDS. So my original question is "answered" (can WDS bundles run standalone? yes) but also leads to a new question:

**Is there a way to disable WDS/HMR handling of a configuration or entrypoint?**

In effect, **I want a local standalone production-ready build of the library** alongside the WDS docs pipeline without having to run two separate commands/processes.

Docusaurus gives us a hook to modify/merge the webpack config; I had to manually patch in MultiCompiler support to test that but I also got that working as an option.

Here is my current config: [docusaurus.config.ts](https://github.com/excaliburjs/Excalibur/blob/67d8c61aa5778371e2cdd92d694d775686eb6fe7/site/docusaurus.config.ts#L106)

I want to believe this is possible but there's probably some special sauce I need to get it working.

Hoping some webpack wizards may be able to point me in the right direction. I felt like it was close with that linked issue, somehow removing the HMR middleware and only applying it to the app entrypoint, but I got lost with how that works.

**Comment by alexander-akait** [comment]
Hello, sorry for delay, you can manually add hot and client entries where you need, our logic here https://github.com/webpack/webpack-dev-server/blob/master/lib/Server.js#L711
